### DISCUSSION TITLE
SBOL 2 to 3 Converter

### DISCUSSION THREAD

**Discussion by PrashantVaidyanathan** [SynBioDex/SBOL-utilities#284] [General]
This is a space for the community to discuss and/or ask questions about the SBOL 2 - SBOL 3 converter

**Comment by Yehuda-Binik** [comment]
Felipe and I were discussing dependency order for the various object converters that still need to be implemented. We have decided that the highest priority at the moment is 
1. `visit_component()`
2.  `visit_sequence_annotation()`

After that, we have not made any decisions in regard to the order we will implement features in. 

The plan right now is for me to finish testing `visit_component()` using the document that Felipe wrote up for me as the basis of a unit test. Meanwhile, Felipe will begin working on `visit_sequence_annotation()`. Once I finish `visit_component()`, Felipe can really get started on `visit_sequence_annotation()`, and we can go from there.

**Reply by fxbuson** [reply]
Here is the unit test file attached and the code that generated it:

```
import sbol2

doc2 = sbol2.Document()

comp = sbol2.ComponentDefinition("comp_1")
comp2 = sbol2.ComponentDefinition("comp_2")
comp3 = sbol2.ComponentDefinition("comp_3")

doc2.add(comp)
doc2.add(comp2)
doc2.add(comp3)

sub_comp = sbol2.Component("sub_comp_1")
sub_comp.definition = comp2.identity

sub_comp2 = sbol2.Component("sub_comp_2", access=sbol2.SBOL_ACCESS_PRIVATE)
sub_comp2.definition = comp3.identity

comp.components = [sub_comp, sub_comp2]

doc2.write("component_test_1.xml")
```
[component_test_1.zip](https://github.com/user-attachments/files/16873023/component_test_1.zip)

**Comment by GeneCodeSavvy** [comment]
Hello, I’m Harsh Sharma from India, and I’m thrilled to express my interest in participating in Google Summer of Code 2025 with your organization. I recently completed my Bachelor’s degree in Biotechnology Engineering and have a strong background in Python, Git, and GitHub workflows.

I’ve been actively contributing to your organization by addressing minor issues in pySBOL3. For example, I resolved the deprecation of the command `python3 setup.py install` by setting up pyproject.toml ([PR](https://github.com/SynBioDex/pySBOL3/pull/458)). Additionally, I’ve made some pylint fixes that are currently being tracked in the Issue here ([Issue](https://github.com/SynBioDex/pySBOL3/issues/433)).

Although I’m still getting familiar with the specification, I’m confident that I could contribute to the pySBOL2 to pySBOL3 converter utility. Please guide me towards any specific issue that you’d like me to work on.

I was advised by @tcmitchell to reach out to @bbartley and @Gonza10V.
